R/kobo_get_project_codebook.R

```diff
@@ -0,0 +1,26 @@
+#' Functions starting with kobo_ are a set of functions to interact with the datacollection tool kobotoolbox.
+#' Get the XLSForm/Codebook for a kobo project. 
+#' @importFrom magrittr %>%
+#' @importFrom dplyr select
+#' @importFrom httr GET
+#' @importFrom readxl read_xlsx
+#' @importFrom purrr map
+#' @param url URL to kobotoolbox, for example https://kf.kobo.prod.in.noacs.io
+#' @param uid The uid for with the project, get it from kobo_get_project_list function
+#' @param username Username for kobotoolbox
+#' @param password Password for kobotoolbox
+#' @return data A list with a tibble for each sheet in the XLSForm.
+#' @export
+kobo_get_project_codebook <- function(url, uid, username, password) {
+  response <- httr::GET(paste0(url, "/api/v2/assets/", uid, "/deployment/?format=xls"), authenticate(username, password))
```

Comment:
Why do we want to get this as xls and not json or csv? Seems to me like that would avoid the readxl dependency?

Reply:
In kobotoolbox the configuration for the project is in XLSForm-format (an .xlsx file with three sheets) It's possible to export this from the api in json or XML, however, the json request returns the entire deployment, and in there, the survey-data is a bit different from the XLSForm layout. Doing it this way allows it to be imported from this function to the environment as a list of tibbles, but also allow you to very quickly write it back as a file using writexl::write_xlsx(codebook, "XLSForm_codebook.xlsx") if you want to import it to kobo for a new project. I think this is the simplest solution and may be worth the dependency.

Reply:
Got it!